reference-adapters/ifms-adapter/src/main/java/org/egov/service/DisbursementService.java

```diff
@@ -328,4 +334,87 @@ private DisbursementResponse enrichDisbursementResponse(DisbursementRequest disb
 
         return disbursementResponse;
     }
+
+    /**
+     * Method to convert epochDateTimeFormat to LocalDateTime
+     * @param epochTime The epoch time in milliseconds
+     * @return The corresponding LocalDateTime object
+    */
+    private LocalDateTime convertEpochToLocalDateTime(long epochTime) {
+        Instant instant = Instant.ofEpochMilli(epochTime);
+        return LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+    }
+
+    /**
+     * Method to get the financial year of the given date
+     * @param date The date for which the financial year is needed
+     * @return The financial year in the format "YYYY-YYYY"
+     */
+    private String getFinancialYear(LocalDateTime date){
+
+        int year = date.getYear();
+        int month = date.getMonthValue();
+
+        // Check if the date falls in the current or next financial year
+        int startYear, endYear;
+        if (month < 4) { // Before April, so belongs to the previous financial year
+            startYear = year - 1;
+            endYear = year;
+        } else {
+            startYear = year;
+            endYear = year + 1;
+        }
+
+        return startYear + "-" + endYear;
+    }
+
+    /**
+     * This method validates the 90Days scenario and processes the disbursement accordingly
+     * @param paymentInstructionFromDisbursement The payment instruction received from the disbursement
+     * @param requestInfo Information related to the request
+     * @param lastPI The last payment instruction
+     * @param originalPI The original payment instruction
+     * @param disbursementRequest The disbursement request
+     * @param sanctionDetails Details related to sanctions
+     * @return The status of the payment
+     */
+    private PaymentStatus validateAndProcessDisbursement(PaymentInstruction paymentInstructionFromDisbursement, RequestInfo requestInfo, PaymentInstruction lastPI,
```

Comment:
@nmohammednawaz-egov why are you calling the same function multiple places?
you can only return a value as true or false and call the function

Reply:
Sure I will change the return type of method to boolean and return true if it is valid for COR PI Request and false if needs to be created as New PI